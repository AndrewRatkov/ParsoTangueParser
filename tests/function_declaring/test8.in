def str f(int x)

if x == 0
then return "";
fi

if f(x - 1) == "a" * 30
then return f(x - 1);
else return "a" + f(x - 1);
fi

enddef