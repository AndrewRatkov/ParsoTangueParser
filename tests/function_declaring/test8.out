[DEF FUN f(INTEGER x)=>STR]
|   
+-->IF
|   |
|   +--->[INT,==]------->[INT,x]
|   |        |
|   |        +---------->[INT,0]
|   |
|   THEN
|   |
|   +--->[RETURN]--->[STR,""]
|   |
|   FI
|   
+-->IF
    |
    +--->[INT,==]------->[CALL STR f]------->[INT,- ]------->[INT,x]
    |        |                                   |
    |        |                                   +---------->[INT,1]
    |        |
    |        +---------->[STR,* ]------->[STR,"a"]
    |                        |
    |                        +---------->[INT,30]
    |
    THEN
    |
    +--->[RETURN]--->[CALL STR f]------->[INT,- ]------->[INT,x]
    |                                        |
    |                                        +---------->[INT,1]
    |
    ELSE
    |
    +--->[RETURN]--->[STR,+ ]------->[STR,"a"]
    |                    |
    |                    +---------->[CALL STR f]------->[INT,- ]------->[INT,x]
    |                                                        |
    |                                                        +---------->[INT,1]
    |
    FI
